pkg/tsdbctl/query.go

```diff
@@ -66,15 +86,22 @@ func newQueryCommandeer(rootCommandeer *RootCommandeer) *queryCommandeer {
 		},
 	}
 
-	cmd.Flags().StringVarP(&commandeer.to, "end", "e", "", "to time")
-	cmd.Flags().StringVarP(&commandeer.from, "begin", "b", "", "from time")
-	cmd.Flags().StringVarP(&commandeer.output, "output", "o", "", "output format: text,csv,json")
-	cmd.Flags().StringVarP(&commandeer.filter, "filter", "f", "", "v3io query filter e.g. method=='get'")
-	cmd.Flags().StringVarP(&commandeer.last, "last", "l", "", "last min/hours/days e.g. 15m")
-	cmd.Flags().StringVarP(&commandeer.windows, "windows", "w", "", "comma separated list of overlapping windows")
+	cmd.Flags().StringVarP(&commandeer.to, "end", "e", "",
+		"End (maximum) time for the query, as a string containing an\nRFC3339 time string, a Unix timestamp in milliseconds, or\na relative time of the format \"now\" or \"now-[0-9]+[mhd]\"\n(where 'm' = minutes, 'h' = hours, and 'd' = \"days\").\nExamples: \"2018-09-26T14:10:20Z\"; \"1537971006000\";\n\"now-3h\"; \"now-7d\". (default \"now\")")
+	cmd.Flags().StringVarP(&commandeer.from, "begin", "b", "",
+		"Start (minimum) time for the query, as a string containing\nan RFC3339 time, a Unix timestamp in milliseconds, a\nrelative time of the format \"now\" or \"now-[0-9]+[mhd]\"\n(where 'm' = minutes, 'h' = hours, and 'd' = \"days\"), or 0\nfor the earliest time. Examples: \"2016-01-02T15:34:26Z\";\n\"1451748866\"; \"now-90m\"; \"0\". (default = <end time> - 1h)")
+	cmd.Flags().StringVarP(&commandeer.output, "output", "o", formatter.DefaultOutputFormat,
+		"Output format in which to display the query results -\n\"text\" | \"csv\" | \"json\".")
+	cmd.Flags().StringVarP(&commandeer.filter, "filter", "f", "",
+		"Query filter, as an Iguazio Continuous Data Platform\nfilter expression. To reference a metric name from within\nthe query filter, use the \"__name__\" attribute.\nExamples: \"method=='get'\"; \"__name__='cpu' AND os=='win'\".")
+	cmd.Flags().StringVarP(&commandeer.last, "last", "l", "",
+		"Return data for the specified time period before the\ncurrent time, of the format \"[0-9]+[mhd]\" (where\n'm' = minutes, 'h' = hours, and 'd' = days>). When setting\nthis flag, don't set the -b|--begin or -e|--end flags.\nExamples: \"1h\"; \"15m\"; \"30d\" to return data for the last\n1 hour, 15 minutes, or 30 days.")
+	cmd.Flags().StringVarP(&commandeer.windows, "windows", "w", "",
+		"Overlapping windows of time to which to apply the\naggregation functions (if set - see -a|--aggregates),\nstarting from the query's end time, as a comma separated\nlist of integer values (\"[0-9]+\"). The duration of each\nwindow is calculated by multiplying the flag value with\nthe aggregation interval (see -i|--aggregation-interval).\nExample: -w \"1,2\" with -i \"2h\" defines overlapping\naggegration windows of 2 hours and 4 hours.")
```

Comment:
How "1,2" became "2,4" ?
> Example: -w \"1,2\" with -i \"2h\" defines overlapping\naggegration windows of 2 hours and 4 hours.

Reply:
Because as I understood from the README (and verified with you f2f), and as I explain in the help text for the `-w|--windows` flag, the duration of each window is calculated by multiplying the value specified with the flag with the aggregation interval that can be specified with the `-i|--aggregation-interval` flag — so window values of `1` and `2` with an aggregation interval of `2h` should yield window durations of 2h and 4h. (The README gives an example of windows `1,2` with the default aggregation interval of `1h`, but I intentionally didn't want to use an interval of `1`, to clearly demonstrate the multiplication.) Did I misunderstand the behavior?

Reply:
incorrect. if step is smaller than a window - multiple rows will appear in the result-set for that window

Reply:
As we agreed on Slack, please provide an alternative explanation of the flag values (that we can use externally) and a relevant example with a step (aggregation interval) that's not 1h. Thanks.

Reply:
UPDATE: As we agreed today, the overlapping-windows documentation in the PR CLI help is correct.